controller/src/main/java/cern/jarrace/controller/rest/RestConfiguration.java

```diff
@@ -0,0 +1,35 @@
+/**
+ * © Copyright 2016 CERN. This software is distributed under the terms of the Apache License Version 2.0, copied
+ * verbatim in the file “COPYING”. In applying this licence, CERN does not waive the privileges and immunities granted
+ * to it by virtue of its status as an Intergovernmental Organization or submit itself to any jurisdiction.
+ */
+package cern.jarrace.controller.rest;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.context.embedded.EmbeddedServletContainerFactory;
+import org.springframework.boot.context.embedded.jetty.JettyEmbeddedServletContainerFactory;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.PropertySource;
+import org.springframework.core.env.Environment;
+
+/**
+ * Spring {@link Configuration} file for defining Spring REST/MVC related beans
+ *
+ * @author tiagomr
+ */
+
+@Configuration
+@PropertySource("classpath:/server.properties")
+public class RestConfiguration {
+
+    @Autowired
+    Environment environment;
+
+    @Bean
+    public EmbeddedServletContainerFactory embeddedServletContainerFactory() {
+        JettyEmbeddedServletContainerFactory jettyEmbeddedServletContainerFactory = new JettyEmbeddedServletContainerFactory();
+        jettyEmbeddedServletContainerFactory.setPort(environment.getProperty("server.port", Integer.class));
```

Comment:
"server.port" is also hardcoded in AgentRegistrySpawner ... could it be done only once?